hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockDataStreamOutput.java

```diff
@@ -408,7 +408,12 @@ private void executePutBlock(boolean close,
     } else {
       byteBufferList = null;
     }
-    flush();
+    try {
+      CompletableFuture.allOf(futures.toArray(EMPTY_FUTURE_ARRAY)).get();
```

Comment:
We should remove futures from the list if they are completed

Reply:
Good point. I have fixed this.